cmd/fluxd/main.go

```diff
@@ -319,7 +320,13 @@ func main() {
 		}
 	}
 
-	if *sshKeygenDir == "" {
+	// Used to determine if we need to generate a SSH key and setup a keyring
```

Comment:
Is the keyring ever needed unless the git URL uses `ssh` (or the shorthand for it)? I.e., could you just check the scheme.

Reply:
Using this approach is safer as the shorthand errors while parsing it as an URL due to the missing protocol. Which results in the same.